src/sudo/pipeline.rs

```diff
@@ -98,6 +98,17 @@ pub fn run(mut cmd_opts: SudoRunOptions) -> Result<(), Error> {
 
     let pid = context.process.pid;
 
+    // prepare switch of apparmor profile
+    #[cfg(feature = "apparmor")]
+    if let Some(profile) = controls.apparmor_profile {
```

Comment:
Shouldn't we do this inside the *then part* of the "is command resolved" if just below?

This is a highly remote possibility, but technically, if apparmor is enabled *but* the errors do occur, this could leak info about the sudoers file.

Reply:
Correcting myself: no, we are already authenticated at this point.